### PR TITLE
Enable uint8 upsampling 2

### PR DESCRIPTION
**Pull request by r-zenine**
Hi everyone, 

This is a supper small PR to enable `unit8` support for `nearest` up-sampling in `cpu` and `cuda`. 
This works enables us to move forward with the support of 'uint8' images in 'torchvision`. 

See impacted issues : 
https://github.com/pytorch/vision/issues/1375
https://github.com/pytorch/vision/issues/1179#issuecomment-558197607

Note: I wanted to add a unit test to ensure we have the expected behavior. I could not locate the `upsampling` unit tests for `nearest`. I can add the test if you point me to the right location. 

Thanks

### CODE REVIEW

**Comment by r-zenine** [comment]
@fmassa

**Comment by yf225** [comment]
@fmassa Would you like to review this PR? Thanks!

**Comment by r-zenine** [comment]
Hi, 

Any update ?

File: aten/src/ATen/native/cpu/UpSampleKernel.cpp [hunk 1]
```diff
@@ -371,7 +371,7 @@ void upsample_nearest1d_backward_kernel_impl(
     Tensor& grad_input,
     const Tensor& grad_output,
     c10::optional<double> scales_w) {
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(grad_output.scalar_type(), "upsample_nearest1d_backward", [&] {
+  AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::Byte, grad_output.scalar_type(), "upsample_nearest1d_backward", [&] {
```
Comment (fmassa) [inline aten/src/ATen/native/cpu/UpSampleKernel.cpp:374]: I'm not sure if we need backward kernels for `uint8` dtype. @gchanan what did we do for convolutions with `int64` type?

**Review by fmassa — COMMENTED** [review]
Thanks a lot for the PR @r-zenine and sorry for the delay in replying!

I think this looks good to go, I have only one overall comment about if we should include the gradient computation for `uint8` or not. @gchanan what did we do for `int64` tensors?

**Comment by fmassa** [comment]
> Note: I wanted to add a unit test to ensure we have the expected behavior. I could not locate the upsampling unit tests for nearest. I can add the test if you point me to the right location.

Here are some tests for `upsample` (which is now called `interpolate`
https://github.com/pytorch/pytorch/blob/83abd7ffbf3f3fe0d2454a3b2f97bc16ef93d97c/test/test_nn.py#L7134-L7142

**Comment by r-zenine** [comment]
Hi @fmassa, 

Thanks for the feedback. 
Let me then add some tests before we merge the PR.

File: aten/src/ATen/native/cpu/UpSampleKernel.cpp (hunk 1 above)
Comment (r-zenine) [reply aten/src/ATen/native/cpu/UpSampleKernel.cpp:374]: Yes, you are right. Now that you talk about it. It does not make sense to have them. Gradients are not going to be `uint8`. 

Changing that

**Comment by r-zenine** [comment]
Hi @fmassa 
I have made the changes requested. Sorry for the delay. I had a hard time putting time aside to work on this.

Have a good weekend

**Review by fmassa — APPROVED** [review]
LGTM, thanks a lot Ryad!

**Pull request closed without merging**